server/coderadar-webapp/src/main/java/org/wickedsource/coderadar/security/CoderadarSecurityConfiguration.java

```diff
@@ -1,31 +1,60 @@
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+import org.wickedsource.coderadar.security.service.TokenService;
 
 @Configuration
 @EnableWebSecurity
+@EnableGlobalMethodSecurity(prePostEnabled = true)
 public class CoderadarSecurityConfiguration extends WebSecurityConfigurerAdapter {
 
+    @Autowired
+    private UserDetailsService userDetailsService;
+
+    @Autowired
+    private TokenService tokenService;
+
     @Override
-    protected void configure(HttpSecurity http) throws Exception {
-        http.authorizeRequests()
-                .antMatchers("/actuator").hasAnyRole("ADMIN")
-                .anyRequest().permitAll()
-                .and()
-                .httpBasic();
+    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+        // user CoderadarUserDetailService for authentication
+        auth.userDetailsService(userDetailsService).passwordEncoder(getPasswordEncoder());
+    }
 
-        http.csrf().disable();
+    @Bean
+    public PasswordEncoder getPasswordEncoder() {
+        return new BCryptPasswordEncoder();
     }
 
-    @Autowired
-    public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception {
-        auth.inMemoryAuthentication()
-                .withUser("admin").password("admin").roles("ADMIN");
+    @Bean
+    public AuthenticationTokenFilter authenticationTokenFilterBean() throws Exception {
+        return new AuthenticationTokenFilter(tokenService);
     }
 
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.csrf().disable()
+                .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS).and()
+
+                .authorizeRequests()
+
+                // only these endpoints can be called without authentication
+                .antMatchers("/actuator", "/user/auth", "user/registration").permitAll()
+
+                .anyRequest().authenticated();
+
+        // put JSON Wen Token authentication before other ones
```

Comment:
permitAll() on /user/register instead of /user/registration, otherwise you cannot register